platforms/documentation/docs/src/docs/userguide/dep-man/advanced/artifact_resolution.adoc

```diff
@@ -0,0 +1,356 @@
+      "files": [
+        {
+          "name": "kotlin-stdlib-jdk7-1.8.10.jar"
+        }
+      ]
+    },
+    {
+      "name": "jdk8ApiElements",
+      "attributes": {
+        "org.gradle.usage": "java-api",
+        "org.gradle.jvm.version": "8"
+      },
+      "files": [
+        {
+          "name": "kotlin-stdlib-jdk8-1.8.10.jar"
+        }
+      ]
+    }
+  ]
+}
+----
+
+As we can see there are a number of *artifacts* available in the *metadata*:
+
+[cols="1,1"]
+|===
+| Variant | Artifact(s) | Purpose
+
+| `apiElements`
+| `kotlin-stdlib-1.8.10.jar`
+| Standard Kotlin runtime (default).
+
+| `runtimeElements`
+| `kotlin-stdlib-jdk7-1.8.10.jar`
+| Provides additional APIs for Java 7 compatibility.
+
+| `jdk7ApiElements`
+| `kotlin-stdlib-jdk8-1.8.10.jar`
+| Provides additional APIs for Java 8 compatibility.
+
+| `jdk8ApiElements`
+| `kotlin-stdlib-common-1.8.10.jar`
+| Shared standard library for Kotlin Multiplatform.
+|===
+
+In an ideal case, once a *variant* is selected, the *artifact* should be automatically resolved.
+However, there are specific reasons why *artifact* selection still happens _after_ *variant* selection.
+For example, what if the metadata for `org.jetbrains.kotlin:kotlin-stdlib:1.8.10` looked like this:
+
+[source,json]
+.kotlin-stdlib-1.8.10.module
+----
+{
+  "variants": [
+    {
+      "name": "apiElements",
+      "attributes": {
+        "org.gradle.category": "library",
+        "org.gradle.dependency.bundling": "external",
+        "org.gradle.jvm.version": "8",
+        "org.gradle.libraryelements": "jar",
+        "org.gradle.usage": "java-api"
+      },
+      "files": [
+        {
+          "name": "kotlin-stdlib-1.8.10.jar"
+        },
+        {
+          "name": "kotlin-stdlib-1.8.10.tar"
+        }
+      ]
+    }
+  ]
+}
+
+----
+
+How would Gradle know which file to download?
+
+== Artifact Sets
+
+An *artifact set* is a group of *artifacts* that belong to a single *variant* of a *module*.
+
+A single *variant* can include multiple *artifact sets*, each serving a different purpose.
+
+Let's look at the `org.jetbrains.kotlin:kotlin-stdlib:1.8.10` example:
+
+[cols="2,1,~"]
+|===
+| Variant | Set | Artifacts in the Set
+
+| `apiElements`
+| 1
+| jar → The packaged library JAR (`kotlin-stdlib-1.8.10.jar`)
+
+|
+| 2
+| tar → The packaged library TAR (`kotlin-stdlib-1.8.10.tar`)
+
+| `runtimeElements`
+| 1
+| jar → The packaged library JAR (`kotlin-stdlib-1.8.10.jar`)
+
+|
+| 2
+| classes → The compiled .class files (`build/classes/java/main/`)
+
+|
+| 3
+| resources → The associated resource files (`build/resources/main/`)
+
+| `jdk8ApiElements`
+| 1
+| jar → The packaged library JAR (`kotlin-stdlib-jdk8-1.8.10.jar`)
+
+| `jdk7ApiElements`
+| 1
+| jar → The packaged library JAR (`kotlin-stdlib-jdk7-1.8.10.jar`)
+|===
+
+The `apiElements` *variant* of `org.jetbrains.kotlin:kotlin-stdlib:1.8.10` provides two *artifact sets*—`jar`, and `tar`—each representing the same distributable in a different form.
```

Comment:
💅 We probably shouldn't use Kotlin as an example here.  It has changed quite a bit, the stdlib shouldn't be directly depended upon, and KMP complicates things.  We could use the java plugins jar vs. classes.  Or make up an artificial example.

Reply:
I will switch to a different example at a later time.